pmd-java/src/main/resources/category/java/performance.xml

```diff
@@ -1005,23 +1005,10 @@ and buffering.
             <property name="xpath">
                 <value>
                     <![CDATA[
-//PrimaryPrefix/Name
-[ends-with(@Image, '.get') or ends-with(@Image, '.getString')]
-[
-    starts-with(@Image, concat(
-        ancestor::MethodDeclaration//FormalParameter/Type/ReferenceType/ClassOrInterfaceType[pmd-java:typeIs('org.apache.commons.fileupload.FileItem')]/../../..//VariableDeclaratorId/@Name,
-        '.')
-    ) or
-    starts-with(@Image, concat(
-        ancestor::MethodDeclaration//LocalVariableDeclaration/Type/ReferenceType/ClassOrInterfaceType[pmd-java:typeIs('org.apache.commons.fileupload.FileItem')]/../../..//VariableDeclaratorId/@Name,
-        '.')
-    ) or
-    starts-with(@Image, concat(
-        ancestor::ClassOrInterfaceBody/ClassOrInterfaceBodyDeclaration/FieldDeclaration/Type/ReferenceType/ClassOrInterfaceType[pmd-java:typeIs('org.apache.commons.fileupload.FileItem')]/../../..//VariableDeclaratorId/@Name,
-        '.')
-    )
-]
-]]>
+//MethodCall
+    [@MethodName = 'get' or @MethodName = 'getString']
+    [VariableAccess[pmd-java:typeIs('org.apache.commons.fileupload.FileItem')]]
```

Comment:
Does this need to be a VariableAccess? it could be any expression I think

Reply:
`typeIs` doesn't throw anymore when a wrong node is given so you can just do `*[pmd-java:typeIs(...)]`

Reply:
That would catch stuff like

```java
private FileItem getItem() { .... }

private void bar() {
    String result = getItem().getString();
}
```

I can add this....